src/registrar/views/domain_request.py

```diff
@@ -217,8 +217,11 @@ def get(self, request, *args, **kwargs):
         if current_url == self.EDIT_URL_NAME and "id" in kwargs:
             del self.storage
             self.storage["domain_request_id"] = kwargs["id"]
-            self.storage["step_history"] = self.db_check_for_unlocking_steps()
-
+        
+        # refresh step_history to ensure we don't erroneously unlock unfinished
+        # steps just because we visited it
+        self.storage["step_history"] = self.db_check_for_unlocking_steps()
```

Comment:
This seems to work, but it also makes it so that any time you select "Tribal Gov" the election office tab will erroneously show as visited. Same with "About your Organization" if you select "Interstate".